### PR TITLE
Align KTO with DPO: Simplify metrics from sum/count to direct averages

### PR DESCRIPTION
**Pull request by albertvillanova**
Align KTO with DPO: Simplify metrics from sum/count to direct averages.

This PR refactors the way training and evaluation metrics are calculated and logged in the `KTOTrainer` class. Instead of accumulating sums and counts for metrics and then averaging them at log time, the code now directly stores per-example averages during metric collection. The logging logic is simplified to compute means of these averages, and obsolete sum/count metric keys are removed.

Part of:
- #4786

### Changes

**Metric calculation and logging refactor:**

* Changed metric accumulation in both `_compute_loss` and `_compute_loss_liger` to store per-example averages (e.g., `rewards/chosen`, `logps/chosen`, `logits/chosen`) instead of sums and counts.
* Removed tracking of explicit count metrics (`count/chosen`, `count/rejected`) and sum metrics (`*_sum`) from the metrics dictionary. 

**Logging logic simplification:**

* Refactored the `log` method to directly average the stored per-example metrics, compute reward margins, and apply the `eval_` prefix for evaluation metrics. The method no longer computes averages from sums and counts, resulting in simpler and more maintainable code.

---

> [!NOTE]
> **Medium Risk**
> Changes how KTO training/eval metrics are aggregated and logged; dashboards or scripts that relied on old `*_sum` / `count/*` keys will see different names and possibly slightly different averaging semantics across logging steps.
> 
> **Overview**
> **KTOTrainer** now records **per-batch averages** for chosen/rejected rewards, logps, and logits (keys like `rewards/chosen` instead of `*_sum` plus `count/*`), dividing gathered totals by `all_num_chosen` / `all_num_rejected` in both `_compute_loss` and `_compute_loss_liger`.
> 
> **Logging** is rewritten to match **DPOTrainer**: average each metric list, derive `rewards/margins`, prefix keys with `eval_` in eval mode, merge into `logs`, then clear the buffer—no more end-of-log recomputation from sums and counts.
> 
> **DPOTrainer** only drops a blank line in `log`; behavior is unchanged.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 94696f648e8ba5aac195695b7aae2729b95fd600. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/kto/kto_trainer.py [hunk 1]
```diff
@@ -1171,38 +1167,15 @@ def prediction_step(self, model, inputs, prediction_loss_only, ignore_keys=None)
...
-        # Add averaged stored metrics to logs
-        for key, metrics in self._metrics[train_eval].items():
-            logs[f"{prefix}{key}"] = torch.Tensor(metrics).mean().item()
-        self._metrics[train_eval].clear()
-        return super().log(logs, start_time)
+        mode = "train" if self.model.training else "eval"
+        metrics = {key: sum(val) / len(val) for key, val in self._metrics[mode].items()}  # average the metrics
+        if "rewards/chosen" in metrics and "rewards/rejected" in metrics:
+            metrics["rewards/margins"] = metrics["rewards/chosen"] - metrics["rewards/rejected"]
+        if mode == "eval":
```
Comment (qgallouedec) [inline trl/experimental/kto/kto_trainer.py:1176]:
```suggestion

        # This method can be called both in training and evaluation. When called in evaluation, the keys in `logs`
        # start with "eval_". We need to add the prefix "eval_" to the keys in `metrics` to match the format.
        if mode == "eval":
```

File: trl/experimental/kto/kto_trainer.py [hunk 2]
```diff
@@ -1171,38 +1167,15 @@ def prediction_step(self, model, inputs, prediction_loss_only, ignore_keys=None)
...
+            metrics = {f"eval_{key}": val for key, val in metrics.items()}
+        logs.update(metrics)
```
Comment (qgallouedec) [inline trl/experimental/kto/kto_trainer.py:1178]:
```suggestion

        logs.update(metrics)
```

**Review by qgallouedec — APPROVED** [review]
lgtm, just a couple of suggestion to align even more
